hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/MiniDFSCluster.java

```diff
@@ -2775,14 +2775,23 @@ public void waitActive(int nnIndex) throws IOException {
     DFSClient client = new DFSClient(addr, conf);
 
     // ensure all datanodes have registered and sent heartbeat to the namenode
-    while (shouldWait(client.datanodeReport(DatanodeReportType.LIVE), addr)) {
-      try {
+    int failedCount = 0;
+    try {
+      while (shouldWait(client.datanodeReport(DatanodeReportType.LIVE), addr)) {
         LOG.info("Waiting for cluster to become active");
         Thread.sleep(100);
-      } catch (InterruptedException e) {
       }
+    } catch (IOException e) {
+      failedCount++;
+      // Cached RPC connection to namenode, if any, is expected to fail once
+      if (failedCount > 1) {
+        LOG.warn("Tried waitActive() " + failedCount
+            + " time(s) and failed, giving up.  " + StringUtils
+            .stringifyException(e));
+        throw e;
+      }
+    } catch (InterruptedException e) {
     }
```

Comment:
How about re-throwing as IOE because ultimately main thread getting interrupted is going to be problematic? `throw new IOException(e)`
Even without this patch, catching `InterruptedException` was no-op but I believe re-throwing it as IOException might help us analyze any hidden bug underneath. Thoughts?

Reply:
Logically made sense, so I have changed it, In general I don't expect any test to fail due to this, but if any will revert to normal, considering minidfs is used by a lot of downstream projects as well